### PR TITLE
ForwardingTopology settings are obsoleted

### PR DESCRIPTION
**Pull request by SeanFeldman**
Connects to https://github.com/Particular/NServiceBus.AzureServiceBus/issues/524

`ForwardingTopology` settings are obsoleted in 7.2.4 and will be removed in 8.0.0
In order not to confuse users that are about to start with the transport, we will not show these two at all. If they try to use it, they will get code warning.

@Particular/azure-maintainers is this still aligned with what we agreed upon?

### CODE REVIEW

**Comment by danielmarbach** [comment]
@SeanFeldman Probably the easiest way to go about this (discussed with @SimonCropp ) is to remove the doco as you did but have it mentioned in the v8 upgrade guide. I added a draft commit to this PR to illustrate it.

**Comment by SeanFeldman** [comment]
@danielmarbach in the upgrade guide we can't really demonstrate the old API as upgrade guide will be using the new API (v8). Aside from showing what was possible via commented out code, not sure what else we can do.

@SimonCropp if you have an idea how to demonstrate the old code in the upgrade guide other than commented out code, please chime in. Thanks.

**Comment by SeanFeldman** [comment]
@SimonCropp please review this PR. Thank you.

File: nservicebus/upgrades/asb-7to8.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+---
+
+
+## [Forwarding Topology](/nservicebus/azure-service-bus/topologies/) number of entities in bundle removed
+
+In Versions 8 and above the API to configure bundle prefix and number of entities in a bundle has been removed:
+
+snippet: 7to8-number-of-entities-bundle
+
+The default number of entities in a bundle is set to one. For existing endpoints which have been running with multiple entities in a bundle, the transport automatically picks up previously configured entities. The default topic name for bundle is set to `bundle-1`.
```
Comment (SimonCropp) [inline nservicebus/upgrades/asb-7to8.md]: perhaps `bundle is now set to one`?

Comment (SimonCropp) [inline nservicebus/upgrades/asb-7to8.md]: "which have been" seems redundant

**Comment by SimonCropp** [comment]
@SeanFeldman upgrade guides can use two versions https://docs.particular.net/nservicebus/upgrades/5to6/moving-away-from-ibus#migrating-away-from-ibus-during-endpoint-creation

**Comment by SeanFeldman** [comment]
Great, that's exactly what I needed. I will also specify the range in v7 that it was possible, since at some point in v7 we've deprecated it.

**Comment by SeanFeldman** [comment]
@SimonCropp have a look at it now. Should be better.

**Comment by SeanFeldman** [comment]
@Particular/azure-maintainers the build if now green. Please review.

File: Snippets/ASB/ASB_8/AddressingStrategies/NamespacePartitioningExtension.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+using NServiceBus;
+using NServiceBus.Configuration.AdvanceExtensibility;
+
+#region custom-namespace-partitioning-strategy-extension
+
+public static class AzureServiceBusNamespacePartitioningStrategyExtensions
+{
+    public static AzureServiceBusNamespacePartitioningSettings MySetting(this AzureServiceBusNamespacePartitioningSettings extensionPoint)
```
Comment (SimonCropp) [inline Snippets/ASB/ASB_8/AddressingStrategies/NamespacePartitioningExtension.cs]: i would say there is no value in returning `AzureServiceBusNamespacePartitioningSettings `

Comment (SeanFeldman) [inline Snippets/ASB/ASB_8/AddressingStrategies/NamespacePartitioningExtension.cs]: We'll be reviewing APIs and probably taking your suggestion in V8 @SimonCropp

Comment (SeanFeldman) [inline Snippets/ASB/ASB_8/AddressingStrategies/NamespacePartitioningExtension.cs]: And V8 cannot be bad
image

Comment (SimonCropp) [inline Snippets/ASB/ASB_8/AddressingStrategies/NamespacePartitioningExtension.cs]: @SeanFeldman in that case LGTM

Comment (SeanFeldman) [inline Snippets/ASB/ASB_8/AddressingStrategies/NamespacePartitioningExtension.cs]: 😆

**Pull request merged**